libclc/ptx-nvidiacl/libspirv/synchronization/aw_barrier.cl

```diff
@@ -0,0 +1,61 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <spirv/spirv.h>
+#include <spirv/spirv_types.h>
+
+_CLC_OVERLOAD _CLC_DEF void __spirv_BarrierInitialize(long *state,

```

Comment:
fixed

Reply:
Thanks for making this change, but I think using the "INTEL" suffix might be worse... These functions aren't part of an Intel extension to SPIR-V.  Tagging a few SPIR-V experts to weigh in on what these functions should be called: @AlexeySachkov, @bashbaug -- any thoughts?

Reply:
@AlexeySotkin, FYI.

Reply:
I think that the naming here should depend on future plans regarding this new extension.

If there are plans to implement it at SPIR-V level as well, then we can go with `__spirv_FunctionNameVENDOR` expecting that there will be an extension specification for instruction `FunctionNameVENDOR`. If we are unsure who would be the author of it or if there will be several companies involved, then we can use `EXT` as vendor string.

If there are no plans to adopt this at SPIR-V level, then I suggest to completely avoid using `__spirv_*` naming scheme to indicate that these are just some compiler built-ins. We could use `__sycl_*` or something else.

Reply:
+1, I was about to write the same thing.  We shouldn't use the `__spirv` naming unless we have plans to define a SPIR-V extension with this functionality.

We should be a bit careful with `__sycl` also though, because not all SYCL implementers will support this functionality either.  Does it make sense to go straight to `__nvvm`, at least until there is a non-CUDA implementation of this feature?

Reply:
How about `__clc`? I have seen some internal helper functions in libclc use this prefix before. I would prefer not to have `__nvvm` make it confusing with CUDA builtins/intrinsics.

Reply:
`__clc` prefix works for me

Reply:
Changed to `__clc`